mobile/views/change-master-password/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef, useState } from "react";
+import {
+  KeyboardAvoidingView,
+  StyleSheet,
+  Platform,
+  TouchableWithoutFeedback,
+  Keyboard,
+  Modal,
+  TextInput as RNTextInput,
+} from "react-native";
+import Alert from "@/components/alert";
+import { ModalView } from "@/components/modal";
+import Text from "@/components/text";
+import Spacer from "@/components/spacer";
+import TextInput from "@/components/textinput";
+import Button from "@/components/button";
+import { selectMasterPassword, useAppSelector, useAppDispatch, changeMasterPassword } from "@/redux";
+
+export type Props = {
+  visible: boolean;
+  onClose: (sucess: boolean) => void;
+};
+
+const ChangeMasterPassword = ({ visible, onClose }: Props) => {
+  const [loadingMasterPassword, setLoadingMasterPassword] = useState(false);
+  const [password, setPassword] = useState("");
+  const [reenterPassword, setReenterPassword] = useState("");
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const masterPassword = useAppSelector(selectMasterPassword)
+  const dispatch = useAppDispatch();
+
+  const inputRef = useRef<RNTextInput>(null);
+
+  useEffect(() => {
+    if (visible) {
+      setTimeout(() => {
+        inputRef.current?.focus();
+      }, 200);
+    }
+  }, [visible]);
+
+  const onConfirm = async () => {
+    if (!password) return;
+
+    if (password !== reenterPassword) {
+      setError("Passwords do not match.");
+      return;
+    }
+
+    if (!masterPassword) {
+      setError("Master password not found.");
+      return;
+    }
+
+    try {
+      setLoadingMasterPassword(true);
+      await dispatch(changeMasterPassword({ newPassword: password, masterPassword })).unwrap()
+      onClose(true);
+    } catch (error: any) {
+      setError(error.message);
+    } finally {
+      setLoadingMasterPassword(false);
+    }
+  };
+
+  if (!visible) return null;
+
+  return (
+    <Modal>
+      <KeyboardAvoidingView behavior={Platform.OS === "ios" ? "padding" : "height"} style={styles.container}>
+        <TouchableWithoutFeedback onPress={Keyboard.dismiss} style={{ height: "100%" }}>
+          <ModalView.Content>
+            <ModalView.Header title="Change master password" onClose={() => onClose(false)} />
+            <Text.BodyMedium>Please, enter the new password:</Text.BodyMedium>
+            <Spacer />
+            <TextInput
+              ref={inputRef}
+              placeholder={`New password`}
+              error={error}
+              secureTextEntry={true}
+              onChangeText={setPassword}
+            />
+            <TextInput
+              placeholder={`Reenter password`}
+              error={error}
+              secureTextEntry={true}
+              onChangeText={setReenterPassword}
+            />
```

Comment:
![image](https://github.com/user-attachments/assets/96a98074-36da-4408-ad55-caa774f840e2)

In most apps, to change the password you have to enter the current password again (to make sure that someone else doesn't pick up your phone and change the password.) Do you think we need confirm the current password here?

Reply:
Great idea.